### ISSUE TITLE
Database Media Storage : PDF Logo file not database aware

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#23751]
### Preconditions (*)
1. magento 2.3-develop

### Steps to reproduce (*)
```
Backend
=======
Stores -> Configuration -> Advanced -> System
    Storage Configuration for Media
        Media Storage = Database
        Select Media Database = default_setup
        Synchronize
        Save Config
Stores -> Configuration
    Sales -> Sales
        Invoice and Packing Slip Design
            Logo for PDF Print-outs
                Choose Logo File
                Save Configuration
Catalog -> Products -> Add Product
    Name = Test
    Price = 111
    Quantity = 111
    New Category
        Name = TC
        Parent Category = Default Category
        Create Category
    Save & Close
    
Frontend
========
Purchase test product

Verify logo in database
=======================
www-data@dev:~/dev1$ mysql -e 'select filename,directory from dev1_magento.media_storage_file_storage;'
+----------------------------------+--------------------------+
| filename                         | directory                |
+----------------------------------+--------------------------+
| preview_image_5d2ee48863d5a.jpeg | theme/preview            |
| preview_image_5d2ee4889216e.jpeg | theme/preview            |
| image.jpg                        | sales/store/logo/default |
+----------------------------------+--------------------------+
www-data@dev:~/dev1$

Verify logo in filesystem
=========================
www-data@dev:~/dev1$ find ./pub/media -name *.jpg
./pub/media/sales/store/logo/default/image.jpg
www-data@dev:~/dev1$

Backend
=======
Sales -> Orders
    View Order 00000001
    Invoice
    Submit Invoice
Sales -> Orders
    Select Order 00000001
    Print Invoice
Verify Logo appears at top of pdf

Clear logo from Filesystem
==========================
www-data@dev:~/dev1$ rm -rf pub/media/sales/
www-data@dev:~/dev1$

Backend
=======
Sales -> Orders
    Select Order 00000001
    Print Invoice
Verify Logo appears at top of pdf
```

### Expected result (*)
1. Logo always appears at top of pdf

### Actual result (*)
1. Logo is missing from pdf after the local pub/media is cleared.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18665` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @gwharton, @gwharton. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. 

The fix will be available with the upcoming `2.3.3` release.
